sssom/parsers.py

```diff
@@ -337,8 +337,12 @@ def from_sssom_dataframe(
     """
     prefix_map = _ensure_prefix_map(prefix_map)
 
+    # Need to revisit this solution.
+    # This is to address: A value is trying to be set on a copy of a slice from a DataFrame
     if "confidence" in df.columns:
-        df["confidence"].replace(r"^\s*$", np.NaN, regex=True, inplace=True)
+        df2 = df.copy()
```

Comment:
Why does this solve the issue? is "df" already a slice (subset) of some original data frame?

Reply:
When I do the `replace()`  `inplace`, it performs the operation on a slice and hence a `.copy()` ensures that's not the case.